File: Algorithms/377. Combination Sum IV.rs
impl Solution {
    pub fn combination_sum4(nums: Vec<i32>, target: i32) -> i32 {
        let length = target as usize;
        let mut dp = vec![0; length + 1];
        dp[0] = 1;
        for i in 1..=target {
            for &j in &nums {
                if i - j >= 0 {
                    dp[i as usize] += dp[(i - j) as usize];
                }
            }
        }
        dp[length]
    }
}

File: Algorithms/377. Combination Sum IV.ts
function combinationSum4(nums: number[], target: number): number {
  const dp = new Array(target + 1).fill(0);
  dp[0] = 1;
  for (let i = 1; i <= target; i++) {
    for (const j of nums) {
      if (i - j >= 0) {
        dp[i] += dp[i - j];
      }
    }
  }
  return dp[target];
}
